pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -65,6 +65,7 @@ type KubeASCheck struct {
 	ignoredEvents         string
 	ac                    *apiserver.APIClient
 	oshiftAPILevel        apiserver.OpenShiftAPILevel
+	providerIDByNodename  map[string]string
```

Comment:
Could you use `cache "github.com/patrickmn/go-cache"` like it is used here: https://github.com/DataDog/datadog-agent/blob/856fc4e71a5ab49334de66717db021e479631f69/pkg/util/cache/cache.go

Like this we will have the cache cleanup, and also it will be threadsafe.